src/algorithm/REDUCE_SUM-Cuda.cpp

```diff
@@ -190,11 +192,79 @@ void REDUCE_SUM::runCudaVariantBlock(VariantID vid)
 
 }
 
-void REDUCE_SUM::runCudaVariant(VariantID vid, size_t tune_idx)
+template < size_t block_size >
+void REDUCE_SUM::runCudaVariantOccGS(VariantID vid)
 {
+  const Index_type run_reps = getRunReps();
+  const Index_type ibegin = 0;
+  const Index_type iend = getActualProblemSize();
+
+  auto res{getCudaResource()};
+
+  REDUCE_SUM_DATA_SETUP;
+
   if ( vid == Base_CUDA ) {
 
-    size_t t = 0;
+    Real_ptr dsum;
+    allocData(DataSpace::CudaDevice, dsum, 1);
+
+    constexpr size_t shmem = sizeof(Real_type)*block_size;
+    const size_t max_grid_size = detail::getCudaOccupancyMaxBlocks(
+        (reduce_sum<block_size>), block_size, shmem);
+
+    startTimer();
+    for (RepIndex_type irep = 0; irep < run_reps; ++irep) {
+
+      cudaErrchk( cudaMemcpyAsync( dsum, &m_sum_init, sizeof(Real_type),
+                                   cudaMemcpyHostToDevice, res.get_stream() ) );
+
+      const size_t normal_grid_size = RAJA_DIVIDE_CEILING_INT(iend, block_size);
+      const size_t grid_size = std::min(normal_grid_size, max_grid_size);
+      reduce_sum<block_size><<<grid_size, block_size,
+                  shmem, res.get_stream()>>>( x,
+                                                   dsum, m_sum_init,
```

Comment:
vertical alignment of args would look better here.

Reply:
agreed

Reply:
improved

Reply:
We should go through everything and improve the spacing of kernel launch args. We often have the first kernel argument at the end of the first line with the launch arguments which is confusing.

Reply:
https://github.com/LLNL/RAJAPerf/issues/371